static/js/admin/vue-permissions-editor/tests/unit/staffRoles.spec.js

```diff
@@ -0,0 +1,248 @@
+
+        moxios.wait(() => {
+            let request = moxios.requests.mostRecent();
+            expect(request.config.method).toEqual('put');
+            expect(JSON.parse(request.config.data)).toEqual(response.assigned);
+            done();
+        });
+    });
+
+    it("it adds un-added users and then sends current staff roles to the server", (done) => {
+        let added_user = { principal: 'dean', role: 'canAccess', type: 'new' };
+        let all_users = [...response.assigned, ...[added_user]];
+        wrapper.find('#is-submitting').trigger('click');
+        wrapper.find(staffRolesForm).vm.$emit('add-user', added_user);
+        wrapper.find(staffRolesForm).vm.$emit('username-set', false);
+
+        moxios.wait(() => {
+            let request = moxios.requests.mostRecent();
+            expect(request.config.method).toEqual('put');
+            expect(JSON.parse(request.config.data)).toEqual(all_users);
+            done();
+        });
+    });
+
+    it("displays inherited staff roles", (done) => {
+        moxios.wait(() => {
+            let cells = wrapper.findAll('.inherited-permissions td');
+            expect(cells.at(0).text()).toEqual(response.inherited[0].principal);
+            expect(cells.at(1).text()).toEqual(response.inherited[0].role);
+            done();
+        });
+    });
+
+    it("does not display an inherited roles table if there are no inherited roles", (done) => {
+        moxios.wait(() => {
+            wrapper.setData({
+                current_staff_roles: { inherited: [], assigned: [] }
+            });
+            expect(wrapper.find('p').text()).toEqual('There are no inherited staff permissions.');
+            done()
+        });
+    });
+
+    it("displays assigned staff roles", (done) => {
+        moxios.wait(() => {
+            let cells = wrapper.findAll('.assigned-permissions td');
+            expect(cells.at(0).text()).toEqual(response.assigned[0].principal);
+            // See test in staffRolesSelect.spec.js for test asserting that the correct option is displayed
+            done();
+        });
+    });
+
+    it("adds new assigned roles", (done) => {
+        moxios.wait(() => {
+            wrapper.find(staffRolesForm).vm.$emit('add-user', user_role);
+            expect(wrapper.vm.updated_staff_roles).toEqual(response.assigned.concat([user_role]));
+            done();
+        });
+    });
+
+    it("does not add a new user with roles if user already exists", (done) => {
+        moxios.wait(() => {
+            let current_staff_roles = response.assigned.concat([user_role]);
+            wrapper.setData({
+                updated_staff_roles: current_staff_roles
+            });
+
+            wrapper.find(staffRolesForm).vm.$emit('add-user', user_role);
+            expect(wrapper.vm.updated_staff_roles).toEqual(current_staff_roles);
+            done();
+        });
+    });
+
+    it("removes assigned roles for previously assigned users", (done) => {
```

Comment:
I think this is testing that the user is added to the deleted list, rather than that they are removed?